src/SDK/Resource/ResourceInfo.php

```diff
@@ -69,10 +73,34 @@ public static function defaultResource(): self
                 ResourceConstants::TELEMETRY_SDK_NAME => 'opentelemetry',
                 ResourceConstants::TELEMETRY_SDK_LANGUAGE => 'php',
                 ResourceConstants::TELEMETRY_SDK_VERSION => 'dev',
+                ResourceConstants::SERVICE_NAME => 'unknown_service',
             ]
         ));
     }
 
+    /**
+     * Create resource attributes from OTEL_RESOURCE_ATTRIBUTES key=value comma-separated entries
+     * @see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#specifying-resource-information-via-an-environment-variable
+     */
+    public static function environmentResource(): self
+    {
+        $attributes = [];
+        $string = getenv('OTEL_RESOURCE_ATTRIBUTES');
```

Comment:
A quick search only found c++ as a potentially unsafe implementation of `getenv`. I chose getenv over $_ENV for two reasons: it's already in use where we already retrieve env vars (in the otel exporters), and I believe it's advised not to access superglobals directly.
Do you think it would be worth creating a trait so that all env var handling is consistent? I already found myself duplicating a small function in a couple of places when handling env vars...

Reply:
Yeah, sorry couldn't find it either. Maybe I mixed it up with putenv (which is only used in tests). What you say makes sense. :)